### DISCUSSION TITLE
Are transform-runtime and regenerator-runtime meant to be used together?

### DISCUSSION THREAD

**Discussion by JoshRobertson** [babel/babel#13004] [Q&A] [answered]
Consider a babel config 
```
      presets: [
        [
          "@babel/preset-env",
          {
            useBuiltIns: "entry",
            corejs: { version: "3.9", proposals: true },
            shippedProposals: true,
          },
        ],
        ["@babel/preset-react", { runtime: "automatic" }],
        "@babel/typescript"
      ],

      plugins: ["@babel/transform-runtime"]
```

relevant package.json
```
    "@babel/core": "^7.13.10",
    "@babel/plugin-transform-runtime": "^7.13.10",
    "@babel/preset-env": "^7.13.10",
    "@babel/preset-react": "^7.12.13",
    "@babel/preset-typescript": "^7.13.0",
    "@babel/runtime": "^7.13.10",
     "regenerator-runtime": "^0.13.7",
```

and a top-level polyfill file
```
import "core-js";
import "regenerator-runtime/runtime";
```

is this the right approach for `useBuiltIns: entry` and `@babel/transform-runtime`?

**Comment by nicolo-ribaudo** [comment, marked as answer]
Using both doesn't cause problems, but it's not necessary.

If you use `@babel/plugin-transform-runtime`, you can remove the `regenerator-runtime/runtime` import since it will be automatically loaded from `@babel/runtime` when needed.

**Reply by sag1v** [reply]
@nicolo-ribaudo Thanks for this answer.
Not sure if relevant here or deserves a new question, but what about `core-js` as a dependency? If using `@babel/plugin-transform-runtime` (with the `corejs:3` option set) do you need to have `core-js` installed as dependency or does it baked-in already?

**Reply by nyngwang** [reply]
To @sag1v:

Hi, I can help you.

> If using `@babel/plugin-transform-runtime` (with the `corejs:3` option set) do you need to have `core-js` installed as dependency or does it baked-in already?

No, but you do need to install `@babel/runtime-corejs3` instead of `@babel/runtime`, as instructed by the [docs](https://babeljs.io/docs/babel-plugin-transform-runtime#corejs). By `@babel/runtime-corejs3`, it uses the special [_pure_ corejs](https://github.com/babel/babel/blob/3956c75123e713c5fa1d3279f6f92cfeac290173/packages/babel-runtime-corejs3/package.json#L16) behind the scene(and `regenerator-runtime` is on the next line btw). For reference, see #13698, which links to my StackOverflow Q&A, if you need an example.

I also recommend you read [my comment there](https://github.com/babel/babel/issues/13157#issuecomment-2016064692) if you're new to Babel and have many confusions with it.
